Common/services/NewsService.swift

```diff
@@ -14,179 +14,173 @@ class NewsService: HTTPService {
             return
         }
 
         let path = "popular?page=\(page)"
-        _ = request(method: .GET, path: path, params: nil, success: successBlock, fail: failBlock)
+        _ = request(method: .GET, path: path, params: nil, handler: httpHandler)
     }
 
     func requestRecentNewsWithDate (_ date: Date,
-                                    success: ((_ result: [News]?) -> Void)?,
-                                    fail: ((_ error: NSError) -> Void)?) {
+                                    handler: ((_ result: Result <[News], Error>) -> Void)?) {
 
         let calendar = Calendar.current
         let components = (calendar as NSCalendar).components([.day, .month, .year], from: date)
 
         let datePath = String(format: "%d-%02d-%02d", components.year!, components.month!, components.day!)
 
-        let successBlock: (_ result: Data?, _ response: URLResponse?) -> Void = {(data, response) in
-            guard let d = data else {
-                return
+        let httpHandler: ((Result <Data?, Error>) -> Void) = { result in
+            switch result {
+            case .success(let data):
+                guard let d = data else {
+                    handler?(.success([News]()))
+                    return
+                }
+
+                let res = (try? News.decodeArrayOfNews(from: d)) ?? [News]()
+
+                DispatchQueue.main.async(execute: {
+                    handler?(.success(res))
+                })
+            case .failure(let error):
+                handler?(.failure(error))
             }
-
-            let res = try? News.decodeArrayOfNews(from: d)
-
-            DispatchQueue.main.async(execute: {
-                success?(res)
-            })
-        }
-
-        let failBlock: (_ error: NSError) -> Void = { (e) in
-            DispatchQueue.main.async(execute: {
-                fail?(e as NSError)
-            })
         }
 
         if ProcessInfo.processInfo.arguments.contains("mockRequests") {
             let mockService = MockService()
-            _ = mockService.request(file: "GET-recent",
-                                    success: successBlock,
-                                    fail: failBlock)
+            _ = mockService.request(file: "GET-recent", handler: httpHandler)
             return
         }
 
-        _ = request(method: .GET, path: "latest/\(datePath)", params: nil, success: successBlock, fail: failBlock)
+        _ = request(method: .GET, path: "latest/\(datePath)", params: nil, handler: httpHandler)
     }
 
     func requestTrendingTopicsWithDate (_ date: Date,
                                         count: Int,
-                                        success: ((_ result: TopicList?) -> Void)?,
-                                        fail: ((_ error: NSError) -> Void)?) -> URLSessionDataTask? {
+                                        handler: ((_ result: Result <TopicList?, Error>) -> Void)?) -> URLSessionDataTask? {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/cafeb7b8636cbafd0216eaa2489bb108aa52959c/Common/services/NewsService.swift#L112"></span>Line should be 120 characters or less: currently 124 characters<br />
<code>line_length</code> <code>NewsService.swift:112</code></td>
    </tr>
  </tbody>
</table>